code/framework/object/src/main/java/io/cattle/platform/object/process/ObjectProcessManager.java

```diff
@@ -25,4 +25,8 @@
 
     ExitReason executeProcess(String processName, Object resource, Map<String, Object> data);
 
+    void scheduleProcessInstanceAndPublishEvent(String processName, Object resource, Map<String, Object> data);
```

Comment:
Its async in any case; it has similar functionality to  scheduleProcessInstance/scheduleStandardProcess with the only one difference - it gets scheduled within DeferredUtils context. And it means that the event will get generated and published right away, and the process engine will pick the process and execute it right after receiving the event. As opposed to the regular behavior, when periodic task within process engine scans for scheduled processes